ADR/00xx-related-pipelines.md

```diff
@@ -0,0 +1,103 @@
+
+### Pipelines as Code (PaC)
+
+Pipelines as Code (PaC) will expand its design to include orchestration logic. This orchestrator will enable a plugable scheduler which is capable of identifying the dependent repositories and Tekton definitions (composing the metadata required to specify a component in AppStudio). When a component's Tekton definition indicates a dependency on another component, PaC will be able to combine the defined pipelines so that the resulting Pipeline definition builds images in the proper order.
+
+When identifying the pipeline definitions to merge, PaC will first look at the linked components for the same branch where a build was triggered from. If no matching branches are found then the default branch will be used.
+
+PaC will need to recursively expand the directed acyclic graph (DAG) to ensure that all dependent components are linked.
+
+### Build service
+
+The build service will be able to detect any component pull specs from earlier in a composite pipeline that differs from a previous Snapshot and replace all references of the previous pull spec with the new pull spec before building the component.
+
+### Integration service
+
+The integration service will be able to support updating multiple Component updates from a single PipelineRun such that all components built in a composite pipeline can be tested together as an Application in any configured integration tests. In order to support this, the composite pipeline constructed by PaC will need to expose multiple results where integration-service can find the pullspecs of the images.
+
+### Eventual consistency
+
+This flow of related changes as proposed in this ADR is limited to changes in the pull request Tekton definitions. It enables related components to be rebuilt as needed to ensure that image references are maintained and up to date. Once pull requests are merged for one component and the resulting container image is built, the resulting Snapshot will be "out of sync" in that some references in dependent images (i.e. RelatedImages in bundles or image references in a Dockerfile) will not match the latest artifact in the Snapshot. When build dependencies within the same Snapshot are enabled ([RHTAP-967](https://issues.redhat.com/browse/RHTAP-967)), this synchronization issue will be detected and relevant policies will fail.
+
+After a new Snapshot is produced, a Renovatebot instance will be responsible for propagating the updated image references to all other Components within the Application. After these pull requests have all been merged, consistency should be achieved again and all relevant policies should pass. 
```

Comment:
How would integration service know what it could try to release? or would it trigger an attempted release since code was merged, and fail until the other components are built and included in a releasable snapshot?